Utils/Dataflow/030_PDFAnalyzer/manager.py

```diff
@@ -1436,7 +1384,8 @@ def show_paper_datasets(self, window, paper):
                     lbl = Tkinter.Label(frame, text=k, font=HEADING_FONT)
                     lbl.grid(row=r)
                     r += 1
-#                    for [d, special] in paper.datasets[k]:
+                    # for [d, special] in paper.datasets[k]:
+                    # special is not needed. Maybe temporary.
                     for d in paper.datasets[k]:
```

Comment:
It still contains two elements, but the second one is only used when finding datasets in interactive mode (same function, line ~1379).
Yes, there is a commented piece of code which discards the second element, in Manager.update_paper_parameter().
Both snippets of code were replaced with more simple ones per decision, not because they were used to look into some kind of possibilities.

Reply:
Sorry, my question was incorrect -- I meant to ask if `datasets[k][i]` contains single or pair of elements.

Checked via debug printing -- single element, indeed:
```
paper.datasets[k]: ['mc10_7TeV.105608.pythiaContactInt_Lambda7_minus.merge.AOD.e574_s933_s946_r1831_r1700', 'mc10_7TeV.105863.pythiaContactInt_Lambda3_minus.merge.AOD.e574_s933_s946_r1831_r1700', 'mc10_7TeV.105864.pythiaContactInt_Lambda4_minus.merge.AOD.e574_s933_s946_r1831_r1777', 'mc10_7TeV.105865.pythiaContactInt_Lambda5_minus.merge.AOD.e574_s933_s946_r1831_r1700', 'mc10_7TeV.105866.pythiaContactInt_Lambda2_plus.merge.AOD.e574_s933_s946_r1831_r1700', 'mc10_7TeV.105867.pythiaContactInt_Lambda3_plus.merge.AOD.e574_s933_s946_r1831_r1700', 'mc10_7TeV.105868.pythiaContactInt_Lambda4_plus.merge.AOD.e574_s933_s946_r1831_r1700', 'mc10_7TeV.105869.pythiaContactInt_Lambda5_plus.merge.AOD.e574_s933_s946_r1831_r1700']
```

And found the piece of code you\`re talking about:
https://github.com/PanDAWMS/dkb/blob/adeff5735726bfaac2fd65cd6a1a992be839510d/Utils/Dataflow/030_PDFAnalyzer/manager.py#L1236-L1240

Thank you! Now I believe I got it.